### ISSUE TITLE
Ruff does not complain about missing tzinfo when using `datetime.datetime.combine`

### ISSUE THREAD

**Issue by RuurdBijlsma** [astral-sh/ruff#19070]
### Summary

Example when leaving out tzinfo when constructing a datetime normally:

```python
1 file reformatted, 87 files left unchanged
src/dynamic_dimensioning_logic/logic.py:78:13: DTZ001 `datetime.datetime()` called without a `tzinfo` argument
   |
76 |         self.hello_world.write(output_df=output_df)
77 |         logger.info("posted-hello_world")
78 |         x = datetime.datetime(year=2025, month=6, day=1, hour=0, minute=0, second=0)
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ DTZ001
79 |         logger.info("x", x=x)
   |
   = help: Pass a `datetime.timezone` object to the `tzinfo` parameter

Found 1 error.
```

Now, when I construct a datetime with datetime.datetime.combine:

```python
y = datetime.datetime.combine(
    datetime.date(year=2025, month=6, day=1),
    datetime.time(hour=0, minute=0, second=0)
)
```

I get no error, even though combine does take a tzinfo argument. This cause a bug for me so I'd love a ruff error here.

I have `select =['ALL']` in my ruff.toml, so it's not disabled.

### Version

0.12.1

**Comment by ntBre** [comment]
Thanks, this sounds like a reasonable rule to me. I don't see a corresponding rule in [flake8-datetimez](https://pypi.org/project/flake8-datetimez/), so we would need to decide where to put this functionality. A few options are a new rule in the DTZ namespace, if we think the upstream linter is unlikely to add more rules, a new RUF rule, or as an extension to one of the existing rules.
